console/v1/tests/consoleplugins.console.openshift.io/AAA_ungated.yaml

```diff
@@ -87,3 +87,75 @@ tests:
           i18n:
             loadType: Invalid
       expectedError: "Unsupported value: \"Invalid\""
+    - name: Should be able to create a ConsolePlugin with valid CSP configuration
+      initial: |
+        apiVersion: console.openshift.io/v1
+        kind: ConsolePlugin
+        spec:
+          displayName: foo
+          backend:
+            type: Service
+          contentSecurityPolicy:
+          - directive: ScriptSrc
+            values:
+            - https://script1.com/
+          - directive: ImgSrc
+            values:
+            - https://img1.com/
+      expected: |
+        apiVersion: console.openshift.io/v1
+        kind: ConsolePlugin
+        spec:
+          displayName: foo
+          backend:
+            type: Service
+          contentSecurityPolicy:
+          - directive: ScriptSrc
+            values:
+            - https://script1.com/
+          - directive: ImgSrc
+            values:
+            - https://img1.com/
+    - name: Should throw an error for duplicate CSP directive entries
+      initial: |
+        apiVersion: console.openshift.io/v1
+        kind: ConsolePlugin
+        spec:
+          displayName: foo
+          backend:
+            type: Service
+          contentSecurityPolicy:
+          - directive: ScriptSrc
+            values:
+            - https://script1.com/
+          - directive: ScriptSrc
+            values:
+            - https://script1.com/
+      expectedError: "Duplicate value: map[string]interface {}{\"directive\":\"ScriptSrc\""
```

Comment:
Do you have the complete error here? Looks like it's been truncated? 

This will do a substring match but I'm curious what comes after this

Reply:
@JoelSpeed the whole msg is:
`"ConsolePlugin.console.openshift.io \"test-vgzng\" is invalid: spec.contentSecurityPolicy[1]: Duplicate value: map[string]interface {}{\"directive\":\"ScriptSrc\"}"`

Reply:
Ok, we should update this line then to `spec.contentSecurityPolicy[1]: Duplicate value: map[string]interface {}{\"directive\":\"ScriptSrc\"}` to capture the field name and the end that got truncated

Reply:
Outstanding to fix
```suggestion
      expectedError: "spec.contentSecurityPolicy[1]: Duplicate value: map[string]interface {}{\"directive\":\"ScriptSrc\"}"
```